src/com/yusufcihan/DynamicComponents/DynamicComponents.java

```diff
@@ -444,7 +433,10 @@ public void Remove(String id) {
     */
     @SimpleFunction(description = "Returns the last created component's ID by Create block.")
     public String LastUsedID() {
-        return LAST_ID;
+        List COMPONENT_IDS = Arrays.asList(COMPONENTS.keySet().toArray());
+        int size = COMPONENT_IDS.size();
+
+        return (size > 0 ? COMPONENT_IDS.get(size - 1).toString() : "");
```

Comment:
How about shortening it?

```suggestion
        Object[] COMPONENT_IDS = COMPONENTS.keySet().toArray();
        return (COMPONENT_IDS.length > 0 ? COMPONENT_IDS[COMPONENT_IDS.length - 1].toString() : "");
```

And by the way we already have `java.util.Arrays;` import in the Line 25.

Reply:
Hello, sorry for the last response. I've been very busy these past few days working on other things off the Kodular platform and out of my own comfort zone. I should be free this weekend, so if you can hold off on **2.1.0** until Saturday or Sunday, that would be appreciated.